pipeline/vars/common.groovy

```diff
@@ -435,4 +441,98 @@ def getComposeId(def osVersion) {
     return composeId
 }
 
+def fetchRCJsonFile(def rhcsVersion){
+
+    def defaultFileDir = "/ceph/cephci-jenkins/latest-rhceph-container-info"
+    def rcJsonFile = "RHCEPH-${rhcsVersion}-RC.json"
+    def rcFileExists = sh(returnStatus: true, script: "ls -l ${defaultFileDir}/${rcJsonFile}")
+    def rcMapExisting = null
+    if(rcFileExists == 0){
+        rcMapExisting = jsonToMap("${defaultFileDir}/${rcJsonFile}")
+    }
+    def rcJsonDetails = ["fileExists" : rcFileExists, "rcMapExisting" : rcMapExisting, "fileName" : rcJsonFile]
+    return rcJsonDetails
+}
+
+def postRCRPMCompose(){
+    echo "in postRCRPMCompose"
+    def composeMap = getCIMessageMap()
+    def rhcsVersion = composeMap["compose-id"].substring(7,10).toLowerCase()
+    def rcJsonDetails = fetchRCJsonFile(rhcsVersion)
+    def osVersion = composeMap["compose-id"].substring(11,17)
+    def compose_path = composeMap["compose-path"].replace("/mnt/redhat/", "")
+    def compose_url = "http://download-node-02.eng.bos.redhat.com/${compose_path}".toString()
+
+    def jsonContent
+    if(rcJsonDetails.fileExists == 0 && composeMap["compose-label"] in rcJsonDetails.rcMapExisting.compose_label){
+        def compose_urls = [
+            "compose_id" : composeMap["compose-id"],
+            "compose_url" : compose_url
+        ]
+        rcJsonDetails.rcMapExisting.compose_urls.add(compose_urls)
+        jsonContent = writeJSON returnText: true, json: rcJsonDetails.rcMapExisting
+        postCompose(jsonContent.trim(), rcJsonDetails.fileName)
+        return
+    }
+    def rcMap = [
+        "compose_label" : composeMap["compose-label"],
+        "compose_urls" : [
+            [
+                "compose_id" : composeMap["compose-id"],
+                "compose_url" : compose_url
+            ]
+        ]
+    ]
+    jsonContent = writeJSON returnText: true, json: rcMap
+    postCompose(jsonContent.trim(), rcJsonDetails.fileName)
+}
+
+def postRCContCompose(){
+    def composeMap = readJSON text: "${params.CI_MESSAGE}"
+    def rhcsVersion = composeMap.tag.name.substring(5,8)
+    def rcJsonDetails = fetchRCJsonFile(rhcsVersion)
+    if(rcJsonDetails.fileExists != 0){
+        println "Missing RC Json File."
+        return
+    }
+    def repoDetails = composeMap.build.extra.image
+    def containerImage = repoDetails.index.pull.find({x -> !(x.contains("sha"))})
+    def repoUrl = repoDetails.yum_repourls.find({x -> x.contains("RHCEPH")})
+    def composeId = repoUrl.split("/").find({x -> x.contains("RHCEPH-${rhcsVersion}")})
+    def compose_ids = rcJsonDetails.rcMapExisting.compose_urls.collect{it.compose_id}
+    if(!(composeId in compose_ids)){
+        println "Compose ID mismatch"
```

Comment:
It is better to throw an error as we don't have the complete information.

Reply:
done